### PR TITLE
[Datasets] Replace `iter_tf_batches` with  `to_tf` in docs

### PR DESCRIPTION
**Pull request by bveeramani**
Signed-off-by: Balaji <balaji@anyscale.com>

## Why are these changes needed?

`to_tf` is preferred over `iter_tf_batches`. For context, see https://github.com/ray-project/ray/pull/29028 (tl;dr: `iter_tf_batches` is too boilerplate-y).  

## Related issue number

See https://github.com/ray-project/ray/pull/29028

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/train/train_loop_utils.py [hunk 1]
```diff
@@ -52,8 +52,8 @@ def get_dataset_shard(
 ) -> Optional[Union["Dataset", "DatasetPipeline"]]:
     """Returns the Ray Dataset or DatasetPipeline shard for this worker.
 
-    You should call ``iter_torch_batches()`` or ``iter_tf_batches()`` on this shard
-    to convert it to the appropriate framework-specific data type.
+    You should call ``iter_torch_batches()`` or :meth:`~ray.data.Dataset.to_tf` on this
```
Comment (clarkzinzow) [inline python/ray/train/train_loop_utils.py:55]: Same here.

**Review by clarkzinzow — COMMENTED** [review]
Looking good overall, one main comment on where `ds.to_tf()` should be in the supported output formats table.

**Review by clarkzinzow — APPROVED** [review]
LGTM!

**Review by amogkam — CHANGES_REQUESTED** [review]
Need to also change in the following files?

- dataset_pipeline.rst
- check-ingest.rst
- tune_hvd_keras.py
- resnet50_ray_air.py

https://sourcegraph.com/search?q=context:global+repo:%5Egithub%5C.com/ray-project/ray%24+iter_tf_batches&patternType=standard

**Comment by bveeramani** [comment]
> dataset_pipeline.rst

This should be okay as-is. We only reference `iter_tf_batches` in `autosummary` and `automethod`.

> check-ingest.rst

Fixed.

> tune_hvd_keras.py

Couldn't find any mention of `iter_tf_batches` in this file.

> resnet50_ray_air.py

Looks like we're doing some low-level benchmark stuff here. Not sure how I'd replace `iter_tf_batches` with `to_tf` without fundamentally changing the benchark.

```
def to_tensor_iterator():
            num_steps = 0
            # TODO(swang): Should also set a local shuffle buffer size here and
            # pass the same one to build_tf_dataset to match the tf.data
            # implementation.
            for batch in dataset.iter_tf_batches(
                batch_size=batch_size,
                dtypes=tf.float32,
                local_shuffle_buffer_size=config["shuffle_buffer_size"],
            ):
                yield batch["image"], batch["label"]
                num_steps += 1
            assert (
                num_steps == num_steps_per_epoch
            ), f"expected {num_steps} to equal {num_steps_per_epoch}"
            print_dataset_stats(dataset)
```

**Comment by bveeramani** [comment]
Failing `octopus-tune-tests-and-examples-small` tests are known to be flaky.
```
//python/ray/tune:pbt_dcgan_mnist_func                                   FAILED in 3 out of 3 in 12.4s
  | //python/ray/tune:pbt_dcgan_mnist_trainable                              FAILED in 3 out of 3 in 11.3s
```

**Review by amogkam — APPROVED** [review]

**Review by richardliaw — APPROVED** [review]

**Pull request merged**